src/api.v2/helpers/worker/workSubmit/createWorkerK8s.js

```diff
@@ -7,37 +7,68 @@ kc.loadFromDefault();
 
 const logger = getLogger();
 
+// getAvailablePods retrieves pods not assigned already to an activityID given a selector
+const getPods = async (namespace, statusSelector, labelSelector) => {
+  const k8sApi = kc.makeApiClient(k8s.CoreV1Api);
+
+  const pods = await k8sApi.listNamespacedPod(
+    namespace, null, null, null, statusSelector, labelSelector,
+  );
+  return pods.body.items;
+};
+
+const getAvailablePods = async (namespace, statusSelector) => getPods(namespace, statusSelector, '!experimentId,!run');
+
+const getAssignedPods = async (experimentId, namespace) => {
+// check if there's already a running pod for this experiment
+  const assignedRunningPods = await getPods(namespace, 'status.phase=Running', `experimentId=${experimentId}`);
+  if (assignedRunningPods.length > 0) {
+    return assignedRunningPods;
+  }
+
+  // check if there's already a pending pod for this experiment
+  const assignedPendingPods = await getPods(namespace, 'status.phase=Pending', `experimentId=${experimentId}`);
+  if (assignedPendingPods.length > 0) {
+    return assignedPendingPods;
+  }
+
+  return [];
+};
+
 const createWorkerResources = async (service) => {
   const { sandboxId } = config;
   const { experimentId } = service.workRequest;
   const namespace = `worker-${sandboxId}`;
   const k8sApi = kc.makeApiClient(k8s.CoreV1Api);
 
-  const [assignedPods, unassignedPods] = await Promise.all(
-    [
-      k8sApi.listNamespacedPod(namespace, null, null, null, 'status.phase=Running', `experimentId=${experimentId}`),
-
-      // Look for items without an experimentId or run label. Run is used by the cleanup operator
-      // so this prevents us from scheduling it as a worker by accident.
-      k8sApi.listNamespacedPod(namespace, null, null, null, 'status.phase=Running', '!experimentId,!run'),
-    ],
-  );
+  // // check if there's already a running pod for this experiment
```

Comment:
the comment is not acccurate, because we are also checking for pending pods. Change to `check if there's already a running or pending pod for this experiment` and also remove the double-commenting.